lua/bson_lua_wrapper.c

```diff
@@ -0,0 +1,372 @@
+#include <stdlib.h>
+#include <string.h>
+ 
+#include <lua5.2/lua.h>
+#include <lua5.2/lauxlib.h>
+ 
+#include <bson_object.h>
+
+static int table_to_bson_object(lua_State *L, BsonObject *obj, char *errorMessage);
+static int table_to_bson_array(lua_State *L, BsonArray *arr, char *errorMessage);
+
+static int bson_object_to_table(lua_State *L, BsonObject *obj, char *errorMessage);
+static int bson_element_to_table(lua_State *L, BsonElement *element, char *errorMessage);
+static int bson_array_to_table(lua_State *L, BsonArray *arr, char *errorMessage);
+
+static int lbson_object_to_bytes(lua_State *L) {
+    BsonObject bson;
+    bson_object_initialize_default(&bson);
+    if (!lua_istable(L, 1)) {
+        return luaL_argerror(L, 1, "Arg 1 must be a table");
+    }
+
+    char errorMessage[255];
+    int success = table_to_bson_object(L, &bson, errorMessage);
+    if (success != 0) {
+        bson_object_deinitialize(&bson);
+        return luaL_argerror(L, 1, errorMessage);
+    }
+
+    uint8_t *bytes = bson_object_to_bytes(&bson);
+    if (bytes == NULL) {
+        bson_object_deinitialize(&bson);
+        return luaL_argerror(L, 1, "Map could not be parsed into bson");
+    }
+
+    size_t obj_size = bson_object_size(&bson);
+    bson_object_deinitialize(&bson);
+
+    char *byte_string = malloc(obj_size * sizeof(char));
+    size_t index = 0;
```

Comment:
@jacobkeeler can we put index in for loop? Or even avoid index at all, you can iterate throw bytes. Also please add comment what exactly you doing here.

Reply:
Can't put the index in the for loop in C, the variable has to be declared beforehand. I should be able to just iterate though the array though.